gait3d/gaitanalysis_with_top_cam.py

```diff
@@ -0,0 +1,996 @@
+        translation_vector = self._get_translation_vector(coords_closed_left = corners['MazeCornerClosedLeft']['coords'])
+        best_result = self._evaluate_maze_shape_using_open_corners(corners_and_likelihoods = corners, tolerance = relative_error_tolerance)
+        if best_result['valid']:
+            side_id = best_result['side_id']
+            logs_to_add = {'maze_normalization_based_on': f'MazeCornerClosed{side_id}_and_MazeCornerOpen{side_id}'}
+            conversion_factor = self._get_conversion_factor_px_to_cm(coords_point_a = corners[f'MazeCornerClosed{side_id}']['coords'],
+                                                                     coords_point_b = corners[f'MazeCornerOpen{side_id}']['coords'],
+                                                                     distance_in_cm = 50)
+            rotation_angle = self._get_rotation_angle_with_open_corner(corners = corners,
+                                                                       side_id = best_result['side_id'],
+                                                                       translation_vector = translation_vector,
+                                                                       conversion_factor = conversion_factor)
+        else:
+            logs_to_add = {'maze_normalization_based_on': f'MazeCornerClosedRight_and_MazeCornerClosedLeft'}
+            conversion_factor = self._get_conversion_factor_px_to_cm(coords_point_a = corners['MazeCornerClosedLeft']['coords'],
+                                                                     coords_point_b = corners['MazeCornerClosedRight']['coords'],
+                                                                     distance_in_cm = 4)
+            rotation_angle = self._get_rotation_angle_with_closed_corners_only(corners = corners,
+                                                                               translation_vector = translation_vector,
+                                                                               conversion_factor = conversion_factor)
+        self._add_to_logs(logs_to_add = logs_to_add)
+        return {'translation_vector': translation_vector, 'rotation_angle': rotation_angle, 'conversion_factor': conversion_factor}
+
+        
+    def _get_corner_coords_with_likelihoods(self, df: pd.DataFrame) -> Dict:
+        corner_coords_with_likelihood = {}
+        for corner_marker_id in ['MazeCornerClosedRight', 'MazeCornerClosedLeft', 'MazeCornerOpenRight', 'MazeCornerOpenLeft']:
+            xy_coords, min_likelihood = self._get_most_reliable_marker_position_with_likelihood(df = df, marker_id = corner_marker_id)
+            corner_coords_with_likelihood[corner_marker_id] = {'coords': xy_coords, 'min_likelihood': min_likelihood}
+        return corner_coords_with_likelihood
+
+
+    def _get_most_reliable_marker_position_with_likelihood(self, df: pd.DataFrame, marker_id: str, percentile: float=99.95) -> Tuple[np.array, float]:
```

Comment:
Might become hard to find a good trade-off between when to trust lower likelihoods just because time interval of stationary detection is longer, but might be something to explore further.